neo4j/__init__.py

```diff
@@ -310,6 +310,23 @@ def verify_connectivity(self, **config):
         """
         raise NotImplementedError
 
+    def supports_multi_db(self):
+        """ Check if the server or cluster supports multi-databases.
+        :return: Returns true if the server or cluster the driver connects to supports multi-databases, otherwise false.
+        :rtype: bool
+        """
+        from neo4j.io._bolt4x0 import Bolt4x0
+
+        multi_database = False
+        cx = self._pool.acquire()
+
+        if cx.PROTOCOL_VERSION >= Bolt4x0.PROTOCOL_VERSION and cx.server_info.version_info() >= Version(4, 0, 0):
```

Comment:
In theory, this could be strange if you have a mixed environment of database version but I guess that is really uncommon. Would be nicer if you query the cx instead of checking the version of it.

Reply:
Note, tag this as experimental feature detection.

Reply:
The feature detection could be implemented into the cx to get what it supports.